### PR TITLE
Issue 5211 - Add interaction builder (IB) preview in editor

### LINKED ISSUE #5211: Need a field in the sidebar to change the current content
### Description

Sometimes we are using opacity or transform: scale(0)  for hovers, after that inconveniently change text.
image

### Other plugins or websites examples

_No response_

### How to implement it

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by Fybex**
# Description

https://github.com/maxi-blocks/maxi-blocks/assets/46112160/3d9eee09-6be7-4e15-b4e1-0b0c735e12f2

Fixes #5211 

# How Has This Been Tested?
Tested if new toggle works in editor and interactions work on frontend without changes

# Test checklist

**_ Front/Back Testing _**

-   [ ] Test interaction builder (IB) preview toggle in editor
-   [ ] Check that toggle isn't saved after the editor reload
-   [ ] Test interaction builder (IB) styles on frontend (should behave as usual)

**_ Pre-Code Testing _**

-   [ ] Test interaction builder (IB) preview toggle in editor
-   [ ] Check that toggle isn't saved after the editor reload
-   [ ] Test interaction builder (IB) styles on frontend (should behave as usual)
-   [ ] Check if new webpack settings for js/min files don't create any problems
-   [ ] Check no commented code and no unnecessary imports
-   [ ] Standards of the project have been followed
-   [ ] No errors/warnings on console

# Checklist

-   [X] My code follows the style guidelines of this project
-   [X] I have performed a self-review of my code
-   [X] I have commented my code, particularly in hard-to-understand areas
-   [X] My changes generate no new warnings/errors
-   [ ] I have added/updated tests that prove my fix is effective or that my feature works
-   [ ] New and existing unit tests pass locally with my changes

## Summary by CodeRabbit

- **New Features**
  - Introduced a new `ToggleSwitch` component for toggling preview of interactions.
  - Implemented a new `Relation` class to manage element interactions and styles.

- **Enhancements**
  - Improved relation handling with a `processRelations` function for better interaction management.
  - Enhanced user control with the addition of `ToggleSwitch` in the `RelationControl` component.

- **Bug Fixes**
  - Fixed an issue where changes were not marked as non-persistent in the `RelationControl` component.

- **Refactor**
  - Streamlined code by moving the `getBlockNameFromUniqueID` function to a more appropriate module.
  - Consolidated import statements for better code organization.

- **Documentation**
  - Updated inline comments to reflect changes in relation processing logic.

- **Style**
  - Adjusted styles to accommodate the new `ToggleSwitch` component within the interface.

- **Tests**
  - Added new test cases to cover the functionality of the `Relation` class and `processRelations` function.

- **Chores**
  - Updated webpack configuration to include source map generation for easier debugging.

### CODE REVIEW

File: src/extensions/relations/Relation.js [hunk 1]
```diff
@@ -0,0 +1,1165 @@
+/**
```
Comment (Fybex) [inline src/extensions/relations/Relation.js:1]: We're using this class on editor now as well, so decided to split it and move from `js` folder. Saved history of changes, but gh diff is broken anyway (

Use sth like https://www.diffchecker.com/ to compare difference if needed 👍

**Review by Olekrut — APPROVED** [review]
Great Job!!! 😎

**Review by elzadj — CHANGES_REQUESTED** [review]
Looks good, great work! Need to pull master and merge the code for relations manually:

1. This branch still has this bug for blogs https://github.com/maxi-blocks/maxi-blocks/pull/5263
2. Preview for clicks doesn't work, could be connected to https://github.com/maxi-blocks/maxi-blocks/pull/5279 
Code to test: 
[ib-click-preview.txt](https://github.com/maxi-blocks/maxi-blocks/files/13487489/ib-click-preview.txt)
3. At the same time, if I switch on click / on hover a couple of times, and then disable the preview, it still shows:

[screencast-wordpress.local-2023.11.28-13_44_49.webm](https://github.com/maxi-blocks/maxi-blocks/assets/2401618/3372936a-5aea-4097-adfe-ac8239ba3d6a)

You can see it on the ib-click-preview.txt, or on Pure Image Light PIL-PRO-67 too (the preview is stuck if you switch to on click):

Screenshot from 2023-11-28 13-52-54

4. We also can't use files from /src for production like this, everything should come from /build or /js . 
5. Eval should not be used on frontend in min.js, it's a security issue.

**Comment by elzadj** [comment]
@Fybex I'm going to try to fix it today, ping me if you are online and want to switch to it :+1:

**Comment by elzadj** [comment]
@Olekrut please re-tests it, and preview for 'on click' + blogs bug and IB bugs that we fixed yesterday. All should work.
Do npm run build to generate .min.js first :+1:

**Review by Olekrut — CHANGES_REQUESTED** [review]
is not so important, but
if enable toggle preview IB > then change action > try to use  toggle preview IB again
https://jam.dev/c/81aca75f-d755-421d-a528-b8f9acbea468
else looks ok

**Comment by elzadj** [comment]
Important, because it's just stuck if you change the action while preview is active.
Trying to fix.

**Comment by elzadj** [comment]
Connected issue (same if you add an IB or change Action or Settings - the preview stays the same):

[screencast-wordpress.local-2023.11.29-16_42_14.webm](https://github.com/maxi-blocks/maxi-blocks/assets/2401618/c29e1bda-b1b1-4158-96a8-70fc81d1dc72)

**Comment by elzadj** [comment]
Preview on removing an interaction left to do.

**Review by Olekrut — CHANGES_REQUESTED** [review]
For test used Creative Page CEP-PRO-04 and Dance School Page DSP-PRO-01

1) Check patterns like HOL-PRO-06
Where few triggers influence to one target ( in this case to container`s BG Layer.
After that changing action also broke ability for preview (second part of video_
https://jam.dev/c/3aa80e9f-084a-42f5-a15e-dbbddd7af8dd
2) Check pattern HOL-PRO-53
if enable preview IB on XL and then change BP, sizes from IB dont displaying correct
https://jam.dev/c/cf66d80e-576c-418b-afb0-ef59d65afdc1
3) Not sure that is important, but check PIL-PRO-34
preview works, but animation is dragging a bit, ( with freezes )
https://jam.dev/c/7c9756f4-d5a1-4f56-a954-e4d681efc931
4) Noticed on PIL-PRO-34
Enable preview change smth in IB ( width of arrow for example ) > disable/enable preview  > is displaying correct only after few swaps.
Worst if remove IB and add again
https://jam.dev/c/9fe04821-ef4c-4bce-a439-36fe4851a116
5) PIL-PRO-12 adding IB due preview is enabled
https://jam.dev/c/d522bd7c-ac6f-4c38-a751-25309a483a6e
6) PTML-PRO-106
IB preview does not work for group with button and dividers
https://jam.dev/c/bde9de81-0ccb-4a94-b010-d177d68fc772

**Comment by elzadj** [comment]
ok, fixed the divider and responsive too, phew
It's ready for review

**Review by Olekrut — CHANGES_REQUESTED** [review]
1) noticed for pil-pro-12 ( and weird that only for this case )
 if change action ( not update page after that)  > change some transform value for this IB > action is back
https://jam.dev/c/b6666ed6-a9c6-4d5a-8a2f-b514c6d04266
2)I feel that speed is a bit lower
Else looks great🔥😍

**Review by Olekrut — APPROVED** [review]
🔥

**Pull request merged**